syntax = "proto3";
option cc_generic_services = true;
option java_generic_services = true;

import "common.proto";

package freedata; 

message FreeDataBlock
{
   string name = 1;
   int32 value = 2;
}

message FreeDataBlocks
{
   repeated FreeDataBlock items = 1;
}

service FreeDataService 
{
   rpc Get (common.EmptyMessage) returns (FreeDataBlocks);

   rpc Remove (common.StringMessage) returns (common.OperationResultMessage);
   rpc Set (FreeDataBlock) returns (common.OperationResultMessage);
   rpc Increment (FreeDataBlock) returns (common.OperationResultMessage);
   rpc CheckExpression (common.StringMessage) returns (common.OperationResultMessage);
}

